dev/source/docs/building-setup-windows10.rst

```diff
@@ -1,75 +1,61 @@
 
 .. _building-setup-windows10:
 
-===========================================================
-Setting up the waf Build Environment on Windows10 using WSL
-===========================================================
+================================================================
+Setting up the Build Environment on Windows10 using WSL1 or WSL2
+================================================================
 
 These setup instructions describe how to setup "Bash on Ubuntu on Windows" (aka "Windows Subsystem for Linux") which allows building with waf.
 
 .. image:: ../images/build-on-windows10-top-image.jpg
     :target: ../_images/build-on-windows10-top-image.jpg
 
-Setup steps
------------
+It involves enabling the built-in Windows Linux environment (WSL) in Windows10,installing a compatible Linux OS image, and finally installing the build environment as if it were a normal Linux system.
 
-.. note::
+Upon completion, you will be able to build ArduPilot binaries and run the native ArudPilot SITL simulator, including the MAVProxy developer Ground Control Station. 
 
-    Starting with the Fall Creators Update (Windows 10 version 1803), enabling the Developer Mode is not longer required for using WSL.
+WSL1 vs WSL2
+------------
 
-#. Enable Ubuntu on Windows which includes the following steps (original `How-To Geek's instructions here <http://www.howtogeek.com/249966/how-to-install-and-use-the-linux-bash-shell-on-windows-10/>`__):
+WSL2 is the latest version of the Windows10 Subsystem for Linux. It is many times faster than WSL1 and is therefore much preferred  for building the code in terms of speed. It can also run MAVProxy and native SITL. However, it does not support USB yet, so uploading code or connecting to the autopilot is not possible. Also, it does not support running Realflight on the same or a remote PC.
 
-    .. note::
+Once installed, you can switch between WSL1 or WSL2 versions as you prefer.
 
-        In case of trouble, please refer to the official documentation from Microsoft : `<https://docs.microsoft.com/en-us/windows/wsl/install-win10>`__
+WSL Setup steps
+---------------
+#. First, follow the instructions `here <https://docs.microsoft.com/en-us/windows/wsl/install-win10>`_. 
```

Comment:
It would be worth recommending the user install the Ubuntu 20.04 distro under WSL, as the linked instructions just give a list of Linux distros.

Reply:
done